reactos/drivers/usb/usbhub/fdo.c

```diff
@@ -1112,7 +1110,10 @@ DestroyUsbChildDeviceObject(
         return STATUS_UNSUCCESSFUL;
     }
 
+    DPRINT("Removing device on port %d (Child index: %d)\n", PortId, Index);
+
     /* Remove the device from the table */
+    // is lock needed?
```

Comment:
In theory we should only ever have one concurrent PNP request. So I think the toaster driver only needs a lock to protect against concurrent device controls telling it to add/remove device. Not 100% sure though.